packages/storage-plugin/src/symbols.ts

```diff
@@ -46,14 +46,14 @@ export interface NgxsStoragePluginOptions {
   }[];
 
   /**
-   * Serailizer for the object before its pushed into the engine.
+   * Serializer for the object before its pushed into the engine.
    */
-  serialize?(obj: any): string;
+  serialize?(obj: any, key?: string): string;
 
   /**
    * Deserializer for the object before its pulled out of the engine.
    */
-  deserialize?(obj: any): any;
+  deserialize?(obj: any, key?: string): any;
```

Comment:
In order to maintain backward compatibility we would need to change to this:
```suggestion
  deserialize?(obj: any): any;
  deserialize?(obj: any, key?: string): any;
```
I did a test.
This does not work:
![image](https://user-images.githubusercontent.com/1948265/75609330-35e49a80-5b10-11ea-96e3-2a08caeb4e2e.png)
It gives this error:
![image](https://user-images.githubusercontent.com/1948265/75609340-4432b680-5b10-11ea-9d27-fd678e833c64.png)

But, this works:
![image](https://user-images.githubusercontent.com/1948265/75609305-0a61b000-5b10-11ea-8ddc-961fb2bf9fa0.png)

Reply:
@markwhitfeld Not sure if you can manage to get it work that way. I tried your example and the compiler was unable to see the second method. Maybe I am missing something with my understanding of overloads in Typescript.

https://www.typescriptlang.org/docs/handbook/functions.html#overloads

I used a slightly different approach. I kept the existing serialize/deserialize methods untouched and I added optional beforeSerialize/afterDeserialize methods. With that approach, not breaking existing code is guaranteed and it offers a better separation of concerns. The serialize/deserialize are about how to serialize and deserialize the objects and the beforeSerialize and afterSerialize are about how to alter these objects before of after the serialization itself.

I am waiting on your feedback.

Regarding whether or not you want to include the 'SerializationStrategy' class in the codebase, I will respect your decision. Having the opportunity to extends the functionality thanks to the new beforeSerialize/afterDeserialize is good enough for me. Please keep me updated so I can remove it and adjust the tests.